src/Magnum/Audio/Context.h

```diff
@@ -84,13 +133,221 @@ class MAGNUM_AUDIO_EXPORT Context {
+         */
+        Configuration& setMonoSourcesCount(Int sources) {
+            _monoSources = sources;
+            _changed |= Setting::MonoSources;
+            return *this;
+        }
+
+        /**
+         * @brief Hint for how stereo sources to support
+         *
+         * Returns `-1`, if no hint was set.
+         */
+        Int stereoSourcesCount() const { return _stereoSources; }
+
+        /**
+         * @brief Set hint for how stereo sources to support
+         *
+         * If not set, no hint will given to OpenAL.
+         */
+        Configuration& setStereoSourcesCount(Int sources) {
+            _stereoSources = sources;
+            _changed |= Setting::StereoSources;
+            return *this;
+        }
+
+        /** @brief Rate at which the OpenAL device is refreshed (in Hz) */
+        Int refreshRate() const { return _refreshRate; }
+
+        /**
+         * @brief Set rate at which the OpenAL device is refreshed (in Hz)
+         * @return Reference to self (for method chaining)
+         */
+        Configuration& setRefreshRate(Int rate) {
+            _refreshRate = rate;
+            _changed |= Setting::Refresh;
+            return *this;
+        }
+
+    private:
+
+        friend class Context;
+
+        /* Create a attribute list from this Configuration to pass to
+         * @ref alcCreateContext(). */
+        std::vector<Int> attributes() const;
+
+        Int _frequency;
+        bool _enableHrtf;
+
+        Int _monoSources;
+        Int _stereoSources;
+
+        Int _refreshRate;
+
+        /* Enum for whether a specific setting to track if it was changed or not.
+         * To prevent overwriting user config where the default
+         * value would be set. Settings not listed will allways be forced. */
+        enum class Setting: UnsignedByte {
+            Hrtf = 1 << 0,
+            MonoSources = 1 << 1,
+            StereoSources = 1 << 2,
+            Refresh = 1 << 3
+        };
+        Corrade::Containers::EnumSet<Setting> _changed;
 };
 
+
+/** @hideinitializer
+@brief Assert that given OpenAL extension is supported
+@param extension    Extension name (from @ref Magnum::Audio::Extensions "Extensions"
+    namespace)
+
+Useful for initial checks on availability of required features.
+
+By default, if assertion fails, an message is printed to error output and the
+application aborts. If `CORRADE_NO_ASSERT` is defined, this macro does nothing.
+Example usage:
+@code
+MAGNUM_AUDIO_ASSERT_EXTENSION_SUPPORTED(Extensions::ALC::SOFTX::HRTF);
+@endcode
+
+@see @ref Magnum::Audio::Context::isExtensionSupported() "Context::isExtensionSupported()",
+    @ref MAGNUM_ASSERT_VERSION_SUPPORTED(), @ref CORRADE_ASSERT(),
+    @ref CORRADE_INTERNAL_ASSERT()
+*/
+#ifdef CORRADE_NO_ASSERT
+#define MAGNUM_AUDIO_ASSERT_EXTENSION_SUPPORTED(extension) do {} while(0)
+#else
+#define MAGNUM_AUDIO_ASSERT_EXTENSION_SUPPORTED(extension)                        \
```

Comment:
Since it is the MagnumAudio library, I thought that might be a good idea. `MAGNUM_ASSERT_AUDIO_EXTENSION_SUPPORTED` works too, though.

- [x] Done